### PR TITLE
dont try to downgrade Kubernetes

### LINKED ISSUE #1668: Don't try to downgrade Kubernetes
```sh
~> kops upgrade cluster
Using cluster from kubectl context: example.com

ITEM	PROPERTY		OLD	NEW
Cluster	KubernetesVersion	v1.5.2	v1.4.7

Must specify --yes to perform upgrade
```

### PR DESCRIPTION
**Pull request by sethpollack**
fix #1668

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/1672)

### CODE REVIEW

**Comment by chrislovecnm** [comment]
@k8s-bot ok to test

**Comment by chrislovecnm** [comment]
@justinsb I am reviewing and moved into 1.5.0 milestone.  Please confirm and merge if I LGTM

File: cmd/kops/upgrade_cluster.go [hunk 1]
```diff
@@ -134,21 +135,32 @@ func (c *UpgradeClusterCmd) Run(args []string) error {
...
+	if err != nil {
+		return err
+	}
+
+	v2, err := semver.Parse(ParseVersion(channelKubernetesVersion))
+	if err != nil {
+		return err
+	}
+
+	if v1.Compare(v2) == -1 {
```
Comment (chrislovecnm) [inline cmd/kops/upgrade_cluster.go]: cool ... semver is nice ... should we do a GTE or LTE for readability?

Comment (sethpollack) [inline cmd/kops/upgrade_cluster.go]: Oh, didn't even notice those helpers. I'll push up in a second.

File: cmd/kops/upgrade_cluster.go [hunk 2]
```diff
@@ -302,3 +314,10 @@ func (c *UpgradeClusterCmd) Run(args []string) error {
 
 	return nil
 }
+
+func ParseVersion(s string) string {
```
Comment (chrislovecnm) [inline cmd/kops/upgrade_cluster.go]: missed this ... we have this in our codebase already

**Comment by justinsb** [comment]
This is kind of tricky.  I'm not sure whether we're recommending a minimum version or a version.  If k8s 1.5.3 has a nasty bug on AWS, would we not recommend 1.5.2.

Let me ponder...

(We also separately have https://github.com/kubernetes/kops/issues/1636 .  Background there is that I've added the functionality to have one manifest that can trigger on the version of kubernetes and recommend a new version of kubernetes or a new image.  Test case yaml: https://github.com/kubernetes/kops/blob/master/tests/integration/channel/simple/channel.yaml So we can have a 1.4 AMI and a 1.5 AMI, and we'll recommend the right one for the right version of k8s.  So similarly we could recommend 1.4.8 and 1.5.2 at the same time.)

**Comment by justinsb** [comment]
So I think I covered this when I did the per-kops-version upgrade logic - sorry - it wasn't my intention to steal your thunder @sethpollack : https://github.com/kubernetes/kops/blob/master/cmd/kops/upgrade_cluster.go#L148

Going to close, but if there's something more we should do here just let me know.

And thank you for the PR!

**Comment by sethpollack** [comment]
NP, thanks

**Pull request closed without merging**